src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs

```diff
@@ -44,14 +45,35 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context)
                 continue;
 
             // Nested type
-            if (symbol.ContainingType != null)
+            if (symbol.ContainingType is not null)
                 continue;
 
 #if ROSLYN_4_4_OR_GREATER
             if (symbol.IsFileLocal && context.Options.GetConfigurationValue(location.SourceTree, s_rule.Id + ".exclude_file_local_types", defaultValue: true))
                 continue;
 #endif
 
+            var symbolName = symbol.Name;
+
+            // dotnet_diagnostic.MA0048.excluded_symbol_names
+            var excludedSymbolNames = context.Options.GetConfigurationValue(location.SourceTree, "dotnet_diagnostic." + s_rule.Id + ".excluded_symbol_names", defaultValue: string.Empty);
+            if (!string.IsNullOrEmpty(excludedSymbolNames))
+            {
+                var matched = false;
+
+                var excludedSymbolNamesSplit = excludedSymbolNames.Split('|', StringSplitOptions.RemoveEmptyEntries);
+                foreach (var excludedSymbolName in excludedSymbolNamesSplit)
+                {
+                    if (IsWildcardMatch(symbolName, excludedSymbolName))
```

Comment:
The logic from similar rules is a little bit more complicated. For instance, https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508?WT.mc_id=DT-MVP-5003978#exclude-specific-symbols

> - Symbol name only (includes all symbols with the name, regardless of the containing type or namespace).
> - Fully qualified names in the symbol's [documentation ID format](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/#id-strings). Each symbol name requires a symbol-kind prefix, such as M: for methods, T: for types, and N: for namespaces.

It seems you only handle the first case. You could also support `T:MyNamespace.Dummy*`. Note that you can generate the documentation id using `DocumentationCommentId.CreateDeclarationId`

Reply:
@meziantou done, please review.

Reply:
Good! Can you also update the documentation before I can merge? (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0048.md)

Reply:
Updated the doc